haystack/document_stores/in_memory/document_store.py

```diff
@@ -34,7 +34,7 @@ class InMemoryDocumentStore:
     def __init__(
         self,
         bm25_tokenization_regex: str = r"(?u)\b\w\w+\b",
-        bm25_algorithm: Literal["BM25Okapi", "BM25L", "BM25Plus"] = "BM25Okapi",
+        bm25_algorithm: Literal["BM25Okapi", "BM25L", "BM25Plus"] = "BM25L",
```

Comment:
Why change this? What are the benefits?

Reply:
it's briefly mentioned in the description and still open for discussion.

BM25Okapi has always been the default algorithm, while recently we made `scale_score` default to `False` for this Retriever to behave like the others (https://github.com/deepset-ai/haystack/issues/6271) and start filtering out negative values to prevent the retriever from returning irrelevant documents (https://github.com/deepset-ai/haystack/pull/6717). However this two don't play well together when the algorithm used is Okapi, which is the only one among the algorithms made available by `rank_bm25` that can returns negative values.

This PR effectively reverses the filtering of zeroes and negatives for the default case (`BM25Okapi` + `scale_score=False`). I think this is suboptimal, because users hardly even change the default BM25 algorithm for the retriever and therefore they will keep receiving irrelevant documents in the output when the default parameters are set. So the options were:
1. set again scale_score default to True (reverting the first change)
2. change the default BM25 algorithm (which changes the absolute values of the scores, but in most cases not their order)
3. leave everything as it is and let this Retriever return irrelevant documents when `scale_score=False` (very close to reverting the second change)

I chose 2, but it's possible to choose any of the other options too. Other ideas welcome as well.

Reply:
Ok, sounds good. 👍